app/controllers/provider_interface/provider_users_controller.rb

```diff
@@ -120,5 +175,12 @@ def find_provider_user
     rescue ActiveRecord::RecordNotFound
       render_404
     end
+
+    def find_provider_permissions_model!
+      ProviderPermissions.find_by!(
+        provider_user: find_provider_user,
+        provider: Provider.find(params[:provider_id]),
+      )
```

Comment:
If we change this into the query below, it won't be possible to find the model for permissions the user does not have access to. I think that means we can get rid of `assert_current_user_can_manage_users_for` and associated methods?

```suggestion
      find_provider_user.provider_permissions.find_by!(
        provider_id: params[:provider_id],
        manage_users: true,
      )
```

Reply:
On the surface it appears so, but the difference is subtle. `find_provider_permissions_model!` returns the relationship between the user being edited and a provider, but the `manage_users` check is for the `current_provider_user` and their relationship to the same provider.

Reply:
Ah right!